theme/templates/accounts/profile.html

```diff
@@ -4,65 +4,242 @@
+                                </button>
+                            {% else %}
+                                <fieldset>
+                                    <div class="btn-file btn-update-picture upload-picture"><span class="glyphicon glyphicon-camera"></span>
+                                        <span>&nbsp;&nbsp;Add a photo</span>
+                                        <input class="upload-picture" type="file" name="picture"></div>
+                                </fieldset>
+                            {% endif %}
+                            </div>
                         {% endif %}
-                        </tbody>
-                    </table>
-                    {% endif %}
+
+                        {#        Password      #}
+                        {% if profile_user.id == user.id %}
+                        <div data-page-mode="edit">
+                            <hr>
+                            <div class="profile-picture-controls text-center">
+                                <div><a data-toggle="collapse" data-target="#change_password">Change password</a></div>
+
+                                <div id="change_password" class="collapse text-left">
+                                    <fieldset>
+                                        <label>Password</label>
+                                        <input class="form-control" autocomplete="off" id="id_password1" name="password1"
+                                               placeholder="Password"
+                                               type="password">
+
+                                        <small class="text-muted">Leave blank unless you want to change your password</small>
+                                    </fieldset>
+                                    <fieldset>
+                                        <label>Password (again)</label>
+                                        <input class="form-control" autocomplete="off" id="id_password2" name="password2"
+                                               placeholder="Password (again)" type="password">
+                                    </fieldset>
+                                    <hr>
+                                </div>
+
+                                {#        Account deactivation       #}
+                                <div><a data-toggle="modal" data-target="#deactivate-account-dialog">Deactivate account</a></div>
+                            </div>
+                        </div>
+                        {% endif %}
+                    </div>
+
+                    <div class="col-xs-12 col-sm-6">
+                        <div data-page-mode="view">
+                            {#        Name       #}
+                            {% if profile_user.first_name or profile_user.last_name %}
+                                <h2>{{ profile_user.first_name }} {% if profile.middle_name %}{{ profile.middle_name }}{% endif %} {{ profile_user.last_name }}</h2>
```

Comment:
@hyi Yes I can see the problem. It's in html code. It seems this problem of None can be both for middle name and last name. As it seems from code the signup form does not require the last name to be filled in.